cacheops/invalidation.py

```diff
@@ -18,16 +18,32 @@
 __all__ = ('invalidate_obj', 'invalidate_model', 'invalidate_all', 'no_invalidation')
 
 
+def delete_invalid_caches(conj_keys):
+    for conj_key in conj_keys:
+        i = 0
+        while True:
+            i, cache_keys = redis_client.sscan(conj_key, cursor=i, count=1000)
+            if cache_keys:
+                redis_client.srem(conj_key, cache_keys)
+                redis_client.delete(cache_keys)
+            if i == 0:
+                break
+    redis_client.delete(conj_keys)
```

Comment:
This moves deletion of invalid cache keys from Lua into Python, where it is no longer atomic – but we no longer care, since the keys have already been renamed. It's also much slower – but it'll actually be done asynchronously, without holding up Rover.